code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/AccountOutputFilter.java

```diff
@@ -0,0 +1,60 @@
+package io.cattle.platform.iaas.api.auth.identity;
+
+import io.cattle.platform.api.auth.Identity;
+import io.cattle.platform.core.constants.AccountConstants;
+import io.cattle.platform.core.constants.ProjectConstants;
+import io.cattle.platform.core.model.Account;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.id.IdFormatter;
+import io.github.ibuildthecloud.gdapi.model.Resource;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.response.ResourceOutputFilter;
+
+import javax.inject.Inject;
+
+public class AccountOutputFilter implements ResourceOutputFilter {
+
+    @Inject
+    IdentityManager identityManager;
+
+
+    @Override
+    public Resource filter(ApiRequest request, Object original, Resource converted) {
+        IdFormatter idFormatter = ApiContext.getContext().getIdFormatter();
```

Comment:
You are right it does hit the database for each account. I could instead just create the identity without verifying it and the id i return from the api may be invalid. Take your pick.